packages/EasyDoctrine/src/DBAL/Types/DateTimeImmutableMicrosecondsType.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyDoctrine\DBAL\Types;
+
+use DateTimeImmutable;
+use DateTimeInterface;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\DateTimeImmutableType;
+
+final class DateTimeImmutableMicrosecondsType extends DateTimeImmutableType
+{
+    /**
+     * @var string
+     */
+    public const FORMAT_DB_DATETIME = 'DATETIME(6)';
+
+    /**
+     * @var string
+     */
+    public const FORMAT_DB_TIMESTAMP = 'TIMESTAMP';
+
+    /**
+     * @var string
+     */
+    public const FORMAT_DB_TIMESTAMP_WO_TIMEZONE = 'TIMESTAMP(6) WITHOUT TIME ZONE';
+
+    /**
+     * @var string
+     */
+    public const FORMAT_PHP_DATETIME = 'Y-m-d H:i:s.u';
+
+    /**
+     * @var string
+     */
+    public const TYPE_NAME = 'datetime';
+
+    /**
+     * @param mixed $value
+     *
+     * @throws \Doctrine\DBAL\Types\ConversionException
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if ($value === null) {
+            return $value;
+        }
+
+        if ($value instanceof DateTimeInterface) {
+            return $value->format(self::FORMAT_PHP_DATETIME);
+        }
+
+        throw ConversionException::conversionFailedInvalidType($value, $this->getName(), ['null', 'DateTimeImmutable']);
+    }
+
+    /**
+     * @param mixed $value
+     *
+     * @throws \Doctrine\DBAL\Types\ConversionException
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform)
```

Comment:
Pls rename the `\EonX\EasyDoctrine\DBAL\Types\DateTimeMicrosecondsType::convertToPhpValue` to `convertToPHPValue` too.

Reply:
Done